### PR TITLE
Initial pass at a presence guide

### PR DESCRIPTION
**Pull request by dwightwatson**
I just implemented Presence on a project I'm toying around with after watching some of Chris's demonstrations and reading a couple of articles on it. Some of the stuff I read was a little out of date, so I thought I would try and contribute a more up-to-date guide now that the feature is officially released. 

I've provided some simple code samples for both the server- and client-sides trying to demonstrate just the Presence feature alone. I also recommend looking into token authentication as a more secure way to implement Presence in production.

Thanks for the great feature and the framework, it's been a lot of fun to work with. I hope this pull request will be of use.

Related issue: #497

### CODE REVIEW

**Comment by Gazler** [comment]
The contents of this guide are great. We are making some pretty significant structure changes to the guides at the moment, which means adding new files is a bit of a pain. Once the guides are restructured, I'll be sure to merge this in.

**Comment by BryanJBryce** [comment]
What's the status on this?

**Comment by jeregrine** [comment]
@dwightwatson If you could make sure this is compatible Phoenix 1.3, the major changes on guide structure are over.

**Comment by dwightwatson** [comment]
Cool, I've just gone through and updated the guide for Phoenix 1.3.

File: docs/presence.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+and register it in `lib/hello_web/channels/user_socket.ex`:
+
+```elixir
+defmodule HelloWeb.UserSocket do
+  use Phoenix.Socket
+
+  channel "room:lobby", HelloWeb.RoomChannel
+
+  #
```
Comment (jeregrine) [inline docs/presence.md]: Remove comment

File: docs/presence.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    {:noreply, socket}
+  end
+end
+```
+
+Finally we can use the client-side Presence library provided by Phoenix in order to manage the state and presence diffs that come down the socket. We listen for the initial `presence_state` event fired after joining to get the initial state and the later `presence_diff` events that contain joins and leaves.
+
+We can use the included `Presence.syncState()` and `Presence.syncDiff()` methods to easily handle these events and sync our `presences` variable with the latest state. When we want to use the current presence state we can pass it through `Presence.list()` in order to get each presence individually.
+
+When we want to iterate the users, we use the `Presences.list()` function which takes a presences object and a callback function. We use this to display the users and the number of devices they are online with.
```
Comment (jeregrine) [inline docs/presence.md]: Should say what the callback function does.

**Comment by Gazler** [comment]
@dwightwatson Thanks so much for writing this guide. I have updated it for 1.3 and output the list of users on the page so that people can see it working.

<3<3<3

**Pull request merged**